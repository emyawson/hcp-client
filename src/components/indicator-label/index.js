export * from './indicator-label.component';
