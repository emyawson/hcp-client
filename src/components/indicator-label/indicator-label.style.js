import styled from 'styled-components';
import { path } from 'ramda';

import { borderRadius, colors, fontSize, spacing, transitions } from 'src/core';
import { weight } from 'src/components/fonts/weights';

const colorsByStatus = {
  active: {
    bg: colors.lavender,
    border: colors.quartzBlue,
    text: colors.blueMarine,
  },
  default: {
    bg: colors.silverLight,
    border: colors.silverDark,
    text: colors.grayMedium,
  },
  error: {
    bg: colors.trafficRed,
    border: colors.trafficRed,
    text: colors.white,
  },
};

const getColorByStatus = ({ active, error }, colorProp) =>
  active
    ? path(['active', colorProp], colorsByStatus)
    : error
      ? path(['error', colorProp], colorsByStatus)
      : path(['default', colorProp], colorsByStatus);

const marginDirectionToValues = direction => {
  const marginSize = spacing.two;
  switch (direction) {
    case 'top':
      return `${marginSize} 0 0 0`;
    case 'right':
      return `0 ${marginSize} 0 0`;
    case 'bottom':
      return `0 0 ${marginSize} 0 `;
    case 'left':
      return `0 0 0 ${marginSize}`;
    default:
      return '0';
  }
};

export const IndicatorLabelSpan = styled.span`
  background-color: ${props => getColorByStatus(props, 'bg')};
  border: 1px solid ${props => getColorByStatus(props, 'border')};
  border-radius: ${borderRadius.three};
  color: ${props => getColorByStatus(props, 'text')};
  display: inline-block;
  font-size: ${fontSize.label};
  font-weight: ${weight.bold};
  line-height: 1.1;
  margin: ${props => marginDirectionToValues(props.margin)};
  padding: ${spacing.one} ${spacing.two};
  transition: ${transitions.default};
`;
