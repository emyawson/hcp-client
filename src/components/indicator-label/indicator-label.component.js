import React from 'react';

import { IndicatorLabelSpan } from './indicator-label.style';

export const IndicatorLabel = ({
  active = false,
  error = false,
  text,
  marginDirection = '',
}) => (
  <IndicatorLabelSpan active={active} error={error} margin={marginDirection}>
    {text}
  </IndicatorLabelSpan>
);
