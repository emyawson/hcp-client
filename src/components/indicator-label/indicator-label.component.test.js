import React from 'react';
import { shallow } from 'enzyme';

import { IndicatorLabel } from './indicator-label.component';

describe('Indicator label component', () => {
  it('renders correctly', () => {
    const mockProps = {
      active: true,
      error: true,
      text: 'label',
      marginDirection: 'left',
    };
    const wrapper = shallow(<IndicatorLabel {...mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('sets default props', () => {
    const mockProps = {
      text: 'label',
    };
    const wrapper = shallow(<IndicatorLabel {...mockProps} />);
    expect(wrapper.props().active).toEqual(false);
    expect(wrapper.props().error).toEqual(false);
    expect(wrapper.props().margin).toEqual('');
  });
});
