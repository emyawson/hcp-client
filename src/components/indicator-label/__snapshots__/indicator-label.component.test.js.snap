// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Indicator label component renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    active={true}
    error={true}
    marginDirection="left"
    text="label"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "active": true,
      "children": "label",
      "error": true,
      "margin": "left",
    },
    "ref": null,
    "rendered": "label",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "active": true,
        "children": "label",
        "error": true,
        "margin": "left",
      },
      "ref": null,
      "rendered": "label",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
